src/python/pants/testutil/rule_runner.py

```diff
@@ -73,6 +74,14 @@
 from pants.util.ordered_set import OrderedSet
 
 
+@contextlib.contextmanager
+def _chdir(path: int | str | bytes | os.PathLike[str] | os.PathLike[bytes]):
+    old_cwd = os.getcwd()
+    os.chdir(path)
+    yield
+    os.chdir(old_cwd)
```

Comment:
Had to roll our own:
- https://docs.python.org/3/library/contextlib.html#contextlib.chdir is in Python 3.11
- https://docs.pytest.org/en/6.2.x/monkeypatch.html would be overkill

Reply:
Needs try-finally.

But can use: https://github.com/pantsbuild/pants/blob/f8df1173069d33b265a44d613533509fcb033276/src/python/pants/util/contextutil.py#L207-L215